content/en/docs/concepts/services-networking/dual-stack.md

```diff
@@ -0,0 +1,148 @@
+
+{{< note >}}
+The default address family for your cluster is the address family of the first service cluster IP range configured via the `--service-cluster-ip-range` flag to the kube-controller-manager.
+{{< /note >}}
+
+You can set `.spec.ipFamily` to either:
+
+   * `IPv4`: The API server will assign an IP from a `service-cluster-ip-range` that is `ipv4`
+   * `IPv6`: The API server will assign an IP from a `service-cluster-ip-range` that is `ipv6`
+
+The following Service specification does not include the `ipFamily` field. Kubernetes will assign an IP address (also known as a "cluster IP") from the first configured `service-cluster-ip-range` to this Service.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+```
+
+The following Service specification includes the `ipFamily` field. Kubernetes will assign an IPv6 address (also known as a "cluster IP") from the configured `service-cluster-ip-range` to this Service.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  ipFamily: IPv6
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+```
+
+For comparison, the following Service specification will be assigned an IPV4 address (also known as a "cluster IP") from the configured `service-cluster-ip-range` to this Service.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  ipFamily: IPv4
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+```
+
+### Type LoadBalancer
+
+On cloud providers which support IPv6 enabled external load balancers, setting the `type` field to `LoadBalancer` in additional to setting `ipFamily` field to `IPv6` provisions a cloud load balancer for your Service.
+
+## Egress Traffic
+
+The use of publicly routable and non-publicly routable IPv6 address blocks is accecptable provided the underlaying CNI provider is able to implement the transport. If you have a Pod that uses non-publicly routable IPv6 and want that Pod to reach off-cluster destinations (eg. the public Internet), you must set up IP masquerading for the egress traffic and any replies. The [ip-masq-agent](https://github.com/kubernetes-incubator/ip-masq-agent) is dual-stack aware, so you can use ip-masq-agent for IP masquerading on dual-stack clusters.
+
+## Known Issues
+
+   * IPv6 network block assignment is using the default IPv4 cidr block size (/24)
+   * Kubenet forces IPv4,IPv6 positional reporting of IPs (--cluster-cidr)
+   * Dual-stack will not function if `EndpointSlice` enhancement is enabled.
```

Comment:
```suggestion
   * Dual-stack networking does not function if the `EndpointSlice` feature gate is enabled.
```
(how does it not function? Is IPv4 affected, or just IPv6?)

Reply:
I will not work for either blocks. Dual-stack would need to be integrated into the EndpointSlice enhancement so in 1.16 the enablement of these enhancements is mutually exclusive. Given that both these enhancements are in alpha in 1.16 this was not possible.